iOS-sample-app/Showcases/Screen Views/Modals/ProgrammaticModalPresentingViewController.swift

```diff
@@ -33,7 +33,8 @@ UIViewController, DismissalDelegate, UIAdaptivePresentationControllerDelegate {
         present(modal, animated: true)
     }
     
-    func didDismiss(viewController: UIViewController) {
+    func dismiss() {
+        dismiss(animated: true)
```

Comment:
Sorry for repeating myself, but this is all entangled together ; see my other comments for more context.

You should not be doing a dismiss of the presentingVC when what you want to do is to dismiss the modalVC. You should dismiss in the modal and then trigger this delegate function (which should be called didDismiss, as its purpose is to tell you that the modal dismissed itself).

Reply:
I don't think so. As you can see [here](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621505-dismiss), and in other posts on swift, dismissing a modal is supposed to be done by the presenting vc. When you call `dismiss` on the modal itself, iOS actually just handles the dismissal to the presenting controller anyway.

> The presenting view controller is responsible for dismissing the view controller it presented. If you call this method on the presented view controller itself, UIKit asks the presenting view controller to handle the dismissal.

Reply:
I'm ok with your changes on that specific point.